eos_data_distribution/ndn/base.py

```diff
@@ -115,6 +115,10 @@ def __init__(self, name, face=None):
         self._callbackCount = 0
         self._responseCount = 0
 
+    def setName(self, name):
+        self.name = Name(name)
```

Comment:
I don't really like this -- if we're going the full "a producer owns one name", I feel like that name should be immutable. If we check the prefixes in the future, this makes it possible to do `registerPrefix(nameA)`, then `setName(nameB)`, and now we own "multiple names".

The existence of this was one reason I started ripping out the Producer class, basically, as I felt it was a cheat.